pkg/alertmanager/alertmanager.go

```diff
@@ -286,6 +346,22 @@ func (am *Alertmanager) StopAndWait() {
 	am.wg.Wait()
 }
 
+func (am *Alertmanager) MergePartialExternalState(part *clusterpb.Part) error {
+	return am.state.MergePartialState(part)
+}
+
+func (am *Alertmanager) Position() int {
+	am.positionMtx.Lock()
+	defer am.positionMtx.Unlock()
+	return am.position
+}
+
+func (am *Alertmanager) SetPosition(p int) {
```

Comment:
Question (don't do it, let's just talk about it). Do we need the `SetPosition()`? What if `Alertmanager` was ring-aware and query the ring on demand each time `Position()` is called?

Reply:
I thought about that but my initial impression is that would be too intense on the ring. We'd basically need to do a lookup for every alert that _tries_ to fire for each tenant across all replicas. 

This seemed like a good enough trade-off to test its term of behaviour and see where it puts us (as this a consistency trade-off)

Reply:
> would be too intense on the ring

It's not. The ring is an in-memory structure. You can call `.Get()` as many times as you want (we do it thousands time / sec in the distributor).

Reply:
TIL - This makes perfect sense.

Reply:
Done, resolving this conversation.